### PR TITLE
Redirect to correct URL for evenstream operations

### PR DESCRIPTION
**Pull request by dagnir**
Fix issues related to correctly constructing the redirect URL for eventstream operations:

 - Eventstream operations are only available on the async client
 - Eventstream operations have additional method parameters

## Testing

 - Create a directory `javadoc-root`
 - Unzip the [generated javadoc](https://github.com/aws/aws-sdk-java-v2/blob/c6c9d6a20944495e61af684e4d8161392abf37ca/buildspecs/release-javadoc.yml#L26) JAR into `javadoc-root`
 - Copy the generated `crosslink_redirect.hml` file to `javadoc-root`
 - Start a simple Python web server (serves HTTP on port 8000 by default)
    `python -m http.server`
 - Open a browser and nagivate to `http://localhost:8000/crosslink_redirect.html?uid=s3-2006-03-01&type=SelectObjectContent` and verify that it redirects correctly
 - Also tested other non event streaming operations like S3 `ListBuckets`, bidirectional eventstreaming on Transcribe's `StartStreamTranscription`
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: scripts/doc_crosslinks/crosslink_redirect.html [hunk 1]
```diff
@@ -39,31 +42,68 @@
...
+                        redirectServiceSuffixLink = redirectServiceSuffixLink + clientObjectName + ".html" ;
+                        methodName = camelize(typeName);
                         redirectServiceSuffixLink =
                             redirectServiceSuffixLink +
-                            "#" + methodName +
-                            "(software.amazon.awssdk.services." + serviceName + ".model." + requestObjectName + ")";
+                            "#" +
+                            methodName + requestSignature(uid, serviceName, typeName);
+                    } else {
+                        redirectServiceSuffixLink = redirectServiceSuffixLink + clientObjectName + ".html" ;
```
Comment (joviegas) [inline scripts/doc_crosslinks/crosslink_redirect.html:101]: The else part was not there before , why is it required now ?

Comment (dagnir) [reply scripts/doc_crosslinks/crosslink_redirect.html:101]: Originally, we always use the sync client name for `clientObjectName` regardless of operation. But not, we need to know if `typeName` is async only first, before we know which client to use

File: scripts/doc_crosslinks/generate_cross_link_data.py [hunk 2]
```diff
@@ -25,10 +28,43 @@ def generateDocsMap(apiDefinitionsPath, apiDefinitionsRelativeFilePath):
             apiContent = json.loads(apiDefinition.read())
             if "uid" in apiContent["metadata"].keys():
                 sdks[apiContent["metadata"]["uid"]] = getServiceName
-            clientClass[apiContent["metadata"]["uid"]] = getClientClassNameFromMetadata(apiContent["metadata"])
-
+            clientClassPrefix[apiContent["metadata"]["uid"]] = getClientClassNameFromMetadata(apiContent["metadata"])
+            asyncOnlyOps = {}
+            for op in apiContent['operations'].keys():
```
Comment (joviegas) [inline scripts/doc_crosslinks/generate_cross_link_data.py:33]: Can we move this to separate function something like updateAsyncOperations ?

Comment (dagnir) [reply scripts/doc_crosslinks/generate_cross_link_data.py]: Done

**Review by joviegas — APPROVED** [review]

**Pull request merged**